### PR TITLE
Fix S2259 FP: Support VB.NET static local variables

### LINKED ISSUE #6271: Fix S2259 FP: Support VB.NET static local variables
Reported here: https://community.sonarsource.com/t/vb-net-static-variable-declaration-causing-null-reference-warning/75186

### Description

Variable Data is initialized to Null. The following operation `StaticLocalInitializationSemaphoreOperation` should forget the state for that variable.

### Repro steps

```
Public Function TestDataContains(text as String) As String ' Returning string instead of boolean doesn't change anything
    Static testData As String = LoadTestData()  ' FP
    Return testData.Contains(text)
End Function
```

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fixes #6271

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/StaticLocalInitializationSemaphore.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
+
+internal sealed class StaticLocalInitializationSemaphore : SimpleProcessor<IStaticLocalInitializationSemaphoreOperationWrapper>
+{
+    protected override IStaticLocalInitializationSemaphoreOperationWrapper Convert(IOperation operation) =>
+        IStaticLocalInitializationSemaphoreOperationWrapper.FromOperation(operation);
+
+    protected override ProgramState Process(SymbolicContext context, IStaticLocalInitializationSemaphoreOperationWrapper semaphore) =>
+        context.State.SetSymbolValue(semaphore.Local, null);
```
Comment (costin-zaharia-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/StaticLocalInitializationSemaphore.cs:32]: Would be nice to have a method here `Remove(Symbol)` or `Clear(Symbol)`.

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM! I've added just a small comment.

File: analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/StaticLocalInitializationSemaphore.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/StaticLocalInitializationSemaphore.cs:32]: Noted for the next sprint

Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/StaticLocalInitializationSemaphore.cs:32]: Remove(Symbol)

**Pull request merged**